hack/tools/go.work

```diff
@@ -1,8 +1,8 @@
 // This is a hack, but it prevents go from climbing further and trying to
 // reconcile the various deps across the "real" modules and this one.
 
-go 1.24.0
```

Comment:
also hack/tools/go.mod and hack/tools/golangci-lint/go.mod?

Reply:
I also just noticed that we don't set `godebug` for `hack/tools/golangci-lint/go.mod`

Reply:
🤷‍♂️